src/core/lib/security/credentials/tls/tls_credentials.cc

```diff
@@ -46,6 +44,14 @@ bool CredentialOptionSanityCheck(grpc_tls_credentials_options* options,
     gpr_log(GPR_ERROR, "TLS credentials options is nullptr.");
     return false;
   }
+  if (!options->crl_directory().empty() && options->crl_provider() != nullptr) {
+    gpr_log(GPR_ERROR,
+            "Setting crl_directory and crl_provider not supported. Using the "
+            "crl_provider.");
+    // TODO(gtcooke94) - Maybe return false here. Right now object lifetime of
```

Comment:
So as of right now there is no way for this to _not_ transfer the ownership of the options to the credentials, because all of the errors still end up returning true. The only return false is when `options == nullptr`, in which case it doesn't exist to transfer.

This has us talking about an overall bad design that we've discovered - `  grpc::experimental::TlsChannelCredentialsOptions options;` simply leaks if not passed through then owned by a `TlsCredentials` (it creates a `new c_credentials` in it's constructor which is what leaks).

We don't like that, but it's a broader change to the API to make it okay, so we left the TODO and said for now we'd treat this misconfiguration like other misconfigurations are treated - they get a default and it continues to work.

WDYT?

Reply:
Yeah, okay, I see the problem.  I agree that we need to fix this but that it can be done in a separate PR.